modules/bluetooth/adapter_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bluetooth
+
+import (
+	"testing"
+
+	godbus "github.com/godbus/dbus"
+
+	"barista.run/bar"
+	"barista.run/base/watchers/dbus"
+	"barista.run/outputs"
+	testBar "barista.run/testing/bar"
+)
+
+func init() {
+	busType = dbus.Test
+}
+
+func TestAdapter(t *testing.T) {
+	testBar.New(t)
+	bus := dbus.SetupTestBus()
+	bluez := bus.RegisterService("org.bluez")
+
+	adapterName := "hci0"
+	adapterObjPath := godbus.ObjectPath("/org/bluez/" + adapterName)
+	adapter := bluez.Object(adapterObjPath, "org.bluez.Adapter1")
+	adapter.SetProperties(map[string]interface{}{
+		"Name":         "foo",
+		"Alias":        "foo alias",
+		"Address":      "28:C2:DD:8B:73:8C",
+		"Discoverable": false,
+		"Pairable":     true,
+		"Powered":      true,
+		"Discovering":  false,
+	}, dbus.SignalTypeNone)
+
+	btModule := Adapter(adapterName)
+	btModule.Output(func(i AdapterInfo) bar.Output {
+		state := "OFF"
+		if i.Powered {
+			state = "ON"
+		}
+		return outputs.Textf("%s: %s", i.Name, state)
+	})
+	testBar.Run(btModule)
+
+	testBar.LatestOutput().AssertText([]string{
+		"foo: ON",
+	})
+}
```

Comment:
Optional: add a test for when the adapter is removed and reconnected? It could happen if the adapter is attached via USB or in a docking station.

Reply:
I added a test for the changing 'Powered' state on the adapter.  Not sure if that's what will get triggered when a physical adapter is removed, but that's what changes when I power off the builtin BT adapters.  I don't have an external BT dongle to test with.